platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/resubmitter/DefaultTransactionResubmitter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.resubmitter;
+
+import com.hedera.hapi.platform.event.StateSignaturePayload;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.platform.config.StateConfig;
+import com.swirlds.platform.consensus.EventWindow;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.system.transaction.ConsensusTransactionImpl;
+import com.swirlds.platform.system.transaction.StateSignatureTransaction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A default implementation of {@link TransactionResubmitter}.
+ */
+public class DefaultTransactionResubmitter implements TransactionResubmitter {
+
+    private EventWindow eventWindow;
+    private final long maxSignatureResubmitAge;
+
+    private final TransactionResubmitterMetrics metrics;
+
+    /**
+     * Constructor.
+     */
+    public DefaultTransactionResubmitter(@NonNull final PlatformContext platformContext) {
+        maxSignatureResubmitAge = platformContext
+                .getConfiguration()
+                .getConfigData(StateConfig.class)
+                .maxSignatureResubmitAge();
+
+        metrics = new TransactionResubmitterMetrics(platformContext);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public List<ConsensusTransactionImpl> resubmitStaleTransactions(@NonNull final GossipEvent event) {
+        if (eventWindow == null) {
+            throw new IllegalStateException("Event window is not set");
+        }
+
+        final List<ConsensusTransactionImpl> transactionsToResubmit = new ArrayList<>();
+        for (final ConsensusTransactionImpl transaction : event.getHashedData().getTransactions()) {
```

Comment:
This is worth the null check.

Reply:
This is only showing up as `@Nullable` due to a quirk with how our serialization framework instantiates objects, in practice it will never be null.

As a general rule, we only require null checks for method arguments, and then only if failing to do a null check may result in an error thrown at a much later time. If the method will throw a nullpointer exception without a null check being performed, then the check doesn't buy us much. The important thing is to throw when there is a problem, as opposed to storing a null in a data structure and being surprised at a much later time when it finally causes a problem.